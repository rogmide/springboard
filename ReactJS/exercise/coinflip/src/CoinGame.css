img {
  width: 200px;
  height: 200px;
  border-radius: 50%;
  box-shadow: rgba(50, 50, 93, 0.25) 0px 2px 5px -1px,
    rgba(0, 0, 0, 0.3) 0px 1px 3px -1px;
}

button {
  appearance: none;
  background-color: #ffffff;
  border-radius: 40em;
  border-style: none;
  box-shadow: #adcfff 0 -12px 6px inset;
  box-sizing: border-box;
  color: #000000;
  cursor: pointer;
  display: inline-block;
  font-family: -apple-system, sans-serif;
  font-size: 1.2rem;
  font-weight: 700;
  letter-spacing: -0.24px;
  margin: 0;
  outline: none;
  padding: 1rem 1.3rem;
  quotes: auto;
  text-align: center;
  text-decoration: none;
  transition: all 0.15s;
  user-select: none;
  -webkit-user-select: none;
  touch-action: manipulation;
}

button:hover {
  background-color: #ffc229;
  box-shadow: #ff6314 0 -6px 8px inset;
  transform: scale(1.125);
}

button:active {
  transform: scale(1.025);
}

@media (min-width: 768px) {
  button {
    font-size: 1.5rem;
    padding: 0.75rem 2rem;
  }
}
