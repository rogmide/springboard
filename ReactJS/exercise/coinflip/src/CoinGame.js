import React, { useState } from "react";
import "./CoinGame.css";
import coin_Front from "./coin-Front.png";
import coin_Back from "./coin-Back.png";

const CoinGame = () => {
  const [currentImg, setCurrentImg] = useState();
  const [numFlip, setNumFlip] = useState(0);
  const [front, setFront] = useState(0);
  const [back, setBack] = useState(0);

  const flip = () => {
    let zeroOne = Math.floor(Math.random() * 2);
    if (zeroOne === 0) {
      setCurrentImg(coin_Front);
      setNumFlip(numFlip + 1);
      setFront(front + 1);
    } else {
      setCurrentImg(coin_Back);
      setNumFlip(numFlip + 1);
      setBack(back + 1);
    }
  };

  return (
    <>
      <div>
        <h1>Let's flip a coin!</h1>
        <img src={currentImg} alt="" />
      </div>
      <br></br>
      <button onClick={() => flip()}>flip</button>
      <p>
        Out of {numFlip}, there has been {front} heads and {back} tails{" "}
      </p>
    </>
  );
};

export default CoinGame;
