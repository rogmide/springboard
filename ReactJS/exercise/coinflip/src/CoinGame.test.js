import React from "react";
import { render, fireEvent } from "@testing-library/react";
import CoinGame from "./CoinGame";

beforeEach(function () {
  jest.spyOn(Math, "random").mockReturnValueOnce(0).mockReturnValueOnce(1);
});

afterEach(function () {
  Math.random.mockRestore();
});

// SMOKE TEST
test("should render", () => {
  render(<CoinGame />);
});

// SNAPSHOT TEST
test("should match snapshot", () => {
  const { asFragment } = render(<CoinGame />);
  expect(asFragment()).toMatchSnapshot();
});

it("works when the img is not there", function () {
  const { queryByTestId } = render(<CoinGame />);
  const coin_img = queryByTestId("coin_img");
  const flip_btn = queryByTestId("flip_btn");
  expect(coin_img).toBeNull();

  fireEvent.click(flip_btn);
  const coin_img_there = queryByTestId("coin_img");
  expect(coin_img_there).toHaveAttribute("src", "coin-Front.png");

  fireEvent.click(flip_btn);
  expect(coin_img_there).toHaveAttribute("src", "coin-Back.png");
});
