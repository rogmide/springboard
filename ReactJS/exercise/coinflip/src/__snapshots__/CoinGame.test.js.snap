// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Let's flip a coin!
    </h1>
  </div>
  <br />
  <button
    data-testid="flip_btn"
  >
    flip
  </button>
  <p>
    Out of 0, there has been 0 heads and 0 tails 
  </p>
</DocumentFragment>
`;
